sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,32 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, arg):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
+           >>> from sympy.abc import x, y
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((oo, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> y = Symbol('y')
```

Comment:
`y` is imported twice. Here and above in `from sympy.abc  ...`.
Also `x` is imported twice. Second one overrides the first one.